docusaurus/docs/iOS/components/message-list.md

```diff
@@ -1,3 +1,266 @@
+        didUpdateChannel channel: EntityChange<ChatChannel>
+    ) {
+        let channelUnreadCount = channelController.channel?.unreadCount ?? .noUnread
+        messageListVC.scrollToLatestMessageButton.content = channelUnreadCount
+    }
+
+    open func channelController(
+        _ channelController: ChatChannelController,
+        didChangeTypingUsers typingUsers: Set<ChatUser>
+    ) {
+        guard channelController.areTypingEventsEnabled else { return }
+
+        let currentUserId = channelController.client.currentUserId
+
+        let typingUsersWithoutCurrentUser = typingUsers
+            .sorted { $0.id < $1.id }
+            .filter { $0.id != currentUserId }
+
+        if typingUsersWithoutCurrentUser.isEmpty {
+            messageListVC.hideTypingIndicator()
+        } else {
+            messageListVC.showTypingIndicator(typingUsers: typingUsersWithoutCurrentUser)
+        }
+    }
+}
+```
+
+## UI Customization
+
+You can customize the message list by subclassing the `ChatMessageListVC` and replacing the `Components.default.messageListVC` component.
+
+```swift
+Components.default.messageListVC = CustomMessageListVC.self
+```
+
+<ComponentsNote />
+
+### Message Content View
+In order to change how the messages are rendered, you need to subclass the `ChatMessageContentView` and replace it in the `Components.default.messageContentView`. For more details on how you can customize the message content view, you can take a look at the [Customizing Messages](message.md#customizing-messages) documentation.
+
+You can also set your custom `ChatMessageContentView` in the `ChatMessageListVC.cellContentClassForMessage()` function, this is especially useful if you have multiple instances of `ChatMessageListVC` and each have different `ChatMessageContentView`'s.
+
+```swift
+final class CustomMessageListVC: ChatMessageListVC {
+
+    override func cellContentClassForMessage(at indexPath: IndexPath) -> ChatMessageContentView.Type {
+        CustomChatMessageContentView.self
+    }
+
+}
+```
+As you can see above, by overriding the `cellContentClassForMessage(at:)` function we can change the `ChatMessageContentView` that is used to render the message.
+
+### Message List Layout
+Like any other component in the SDK, you can customize the message list layout by overriding the `setUpLayout()` lifecycle method when subclassing `ChatMessageListVC`.
+
+```swift
+final class CustomMessageListVC: ChatMessageListVC {
+    override func setUpLayout() {
+        super.setUpLayout()
+
+        NSLayoutConstraint.activate([
+            scrollToLatestMessageButton.centerXAnchor.constraint(equalTo: view.layoutMarginsGuide.centerXAnchor)
+        ])
+
+        dateOverlayView.removeFromSuperview()
+    }
+}
+```
+In the simple example above, we moved the `scrollToLatestMessageButton` to the center bottom of message list, instead of the bottom right corner, and also removed the `dateOverlayView` from the view hierarchy.
```

Comment:
```suggestion
In the simple example above, the `scrollToLatestMessageButton` is moved to the center bottom of message list, instead of the bottom right corner, and also the `dateOverlayView` is removed from the view hierarchy.
```